### PR TITLE
feat: Remove duplicate kustomize config

### PR DESCRIPTION
**Pull request by tejal29**
fixes #7658 

In this PR, 
- Removing the `deploy.kustomize` section all together. 

On v1, `kustomize` deployer used to 
- render using `kustomize build` if kustomize binary was present or `kubeclt version` was < v1.14. 
  - if kustomize binary is not present or `kubeclt version` was > v1.14, skaffold would use kubectl kustomize command and pass in the build args.
- deploy using `kubectl` apply. 

Since, kustomize was only used to run `kustomize build` which is now part of render phase, there is no need for kustomize deployer. 

Summary of Changes: 
- remove `kustomizeDeployer` from config
-  remove `pkg/skaffold/deploy/kustomize` package. 
  - Move pkg/skaffold/deploy/kustomize/util.go → pkg/skaffold/render/generate/util.go
  - move all in pkg/skaffold/deploy/kustomize/kustomize_test.go →    pkg/skaffold/deploy/kubectl/kustomize_test.go (except for multiple kustomization dir tests. They pass however are flaky) 
     These tests instead of calling `kustomize.Deploy`, does the follow. 
     - creates a kubectl renderer to render via kustomize build
     - creates a kubectl deployer.
     - Makes sure all tests pass.
- Fixing all unit tests and cleaning up existing usage of `KustomizeDeploy` 
- Removed Kustomize deploy samples from docs/
   - Moved them to samples/renderers. However these are not yet visible.

Following two tests are  removed due to their flaky nature on this PR. 
- https://github.com/GoogleContainerTools/skaffold/blob/v1.39.1/pkg/skaffold/deploy/kustomize/kustomize_test.go#L145
- https://github.com/GoogleContainerTools/skaffold/blob/v1.39.1/pkg/skaffold/deploy/kustomize/kustomize_test.go#L256
On my PR, the order of mock run was not consistent for the following two command
```
CmdRunOut("kustomize build "+tmpDir.Path("a"), kubectl.DeploymentWebYAML).
AndRunOut("kustomize build "+tmpDir.Path("b"), kubectl.DeploymentAppYAML).
```
My initial guess is the render command before calling deploy or cleanup updates the timestamp of these dir. When walking the test directory the order of list is underterministic.  We could look into sorting the list. However, that needed a lot of changes. Hence removed them.

### CODE REVIEW

File: docs/content/en/samples/renderers/kustomize.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+manifests:
```
Comment (aaron-prindle) [inline docs/content/en/samples/renderers/kustomize.yaml:1]: This should be in docs-v2/ dir (not docs/)

File: docs/content/en/docs/pipeline-stages/renderers/kustomize.md [hunk 2]
```diff
@@ -2,36 +2,32 @@
 title: "Kustomize"
```
Comment (aaron-prindle) [inline docs/content/en/docs/pipeline-stages/renderers/kustomize.md:2]: this file should be in docs-v2/ dir (not docs/)

File: pkg/skaffold/render/generate/generate_test.go [hunk 3]
```diff
@@ -136,31 +138,50 @@ func TestGenerate(t *testing.T) {
...
-			// Using "filepath" to join path so as the result can fix when running in either linux or
-			// windows (skaffold integration test).
-			commands: testutil.CmdRun(fmt.Sprintf("kpt fn render fn --output=%v",
-				filepath.Join(".kpt-pipeline", "fn"))),
-			expected: manifest.ManifestList{},
+			useKubectlKustomize: true,
+			commands:            testutil.CmdRunOut("kustomize build base", patchYaml),
+			expected:            manifest.ManifestList{[]byte(patchYaml)},
 		},
+		/*
```
Comment (aaron-prindle) [inline pkg/skaffold/render/generate/generate_test.go:162]: nit: should this test be removed (vs commented)?

File: docs/content/en/samples/renderers/kustomize.yaml (hunk 1 above)
Comment (tejal29) [reply docs/content/en/samples/renderers/kustomize.yaml:1]: Resolved. I had to make additional changes in `TestParseSamples` and all docs-v2/samples

File: pkg/skaffold/render/generate/generate_test.go (hunk 3 above)
Comment (tejal29) [reply pkg/skaffold/render/generate/generate_test.go]: fixed the test

**Review by aaron-prindle — APPROVED** [review]
LGTM!

**Pull request merged**